nrf_security/src/CMakeLists.txt

```
@@ -127,6 +127,7 @@ append_with_prefix(src_tls ${ARM_MBEDTLS_PATH}/library
     ssl_tls13_server.c
     ssl_tls13_client.c
     ssl_tls13_generic.c
+    ssl_debug_helpers_generated.c
 )
```

```suggestion
)

if (CONFIG_MBEDTLS_DEBUG)
append_with_prefix(src_tls ${ARM_MBEDTLS_PATH}/library
ssl_debug_helpers_generated.c
)
endif()
```

Unneccessary source files slow down the build and create noise in the logs.

The whitespace in my proposed patch is not correct.

NB: I know you are following prior-art fairly well. The prior-art is also noisy and slow.